src/stateManager/BlockValidation.ts

```diff
@@ -0,0 +1,15 @@
+import { ValidationResult } from "./types";
+import { ExecutionFlags } from "@/DataTypes";
+
+export async function runPipeline<
+    TContext,
+    TStep extends (context: TContext) => Promise<ValidationResult>
+>(steps: TStep[], context: TContext): Promise<ValidationResult> {
+    for (const step of steps) {
+        const result = await step(context);
+        if (result.executionFlag !== ExecutionFlags.SUCCESS) {
+            return result; // Stop on first non-SUCCESS
+        }
+    }
```

Comment:
I really like this idea, but it doesn't have to be exclusive to blockValidation - the dispute system can also use it and in general all pipelines that use checks. These small steps can also isolate fraud proof detection when deviation is objective and is also useful for subjective deviation like detecting spam. Move this to utils or somewhere more general